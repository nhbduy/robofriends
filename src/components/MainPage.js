import React, { Component } from 'react';

import Header from './Header';
import List from './List';
import Search from './Search';
import Scroll from './Scroll';
import ErrorBoundry from './ErrorBoundry';

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots = () => {
    const { searchField, robots } = this.props;

    return robots.filter(item =>
      item.name.toLowerCase().includes(searchField.toLowerCase())
    );
  };

  render() {
    const { isPending, onSearchChange } = this.props;

    const loadingDOM = <h1 className='tc'>Loading...</h1>;
    const contentDOM = (
      <div className='tc'>
        <Header />
        <Search searchChange={onSearchChange} />
        <Scroll>
          <ErrorBoundry>
            <List data={this.filteredRobots()} />
          </ErrorBoundry>
        </Scroll>
      </div>
    );

    return isPending ? loadingDOM : contentDOM;
  }
}

export default MainPage;
