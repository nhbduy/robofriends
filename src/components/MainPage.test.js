import React from 'react';

import { shallow } from 'enzyme';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  };

  wrapper = shallow(<MainPage {...mockProps} />);
});

it('expect to render MainPage component', () => {
  expect(wrapper.debug()).toMatchSnapshot();
});

it('filters robots correctly', () => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 1,
        name: 'Leanne Graham',
        username: 'Bret',
        email: 'Sincere@april.biz'
      }
    ],
    searchField: 'John',
    isPending: false
  };

  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper.instance().filteredRobots()).toEqual([]);
});

it('filters robots correctly 2', () => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 1,
        name: 'Leanne Graham',
        username: 'Bret',
        email: 'Sincere@april.biz'
      }
    ],
    searchField: 'Leanne',
    isPending: false
  };

  const filteredResult = [
    {
      id: 1,
      name: 'Leanne Graham',
      username: 'Bret',
      email: 'Sincere@april.biz'
    }
  ];

  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper.instance().filteredRobots()).toEqual(filteredResult);
});
