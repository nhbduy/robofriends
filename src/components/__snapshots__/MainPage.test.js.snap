// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage component 1`] = `
"<div className=\\"tc\\">
  <Header />
  <Search searchChange={[undefined]} />
  <Scroll>
    <ErrorBoundry>
      <List data={{...}} />
    </ErrorBoundry>
  </Scroll>
</div>"
`;
